File: rb/features.rb
# StephenKing SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/test_feature'


module StephenKingFeatures
  def self.make_feature(name)
    case name
    when "base"
      StephenKingBaseFeature.new
    when "test"
      StephenKingTestFeature.new
    else
      StephenKingBaseFeature.new
    end
  end
end

File: py/stephenking_sdk/features.py
# StephenKing SDK feature factory

from stephenking_sdk.feature.base_feature import StephenKingBaseFeature
from stephenking_sdk.feature.test_feature import StephenKingTestFeature


def _make_feature(name):
    features = {
        "base": lambda: StephenKingBaseFeature(),
        "test": lambda: StephenKingTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
